File: LeetCode/Python Solutions/Isomorphic Strings/isomorphic.py
class Solution:
    @staticmethod
    def isIsomorphic(s: str, t: str) -> bool:
        l={}
        if len(s)!=len(t):
            return False
        for i in range(len(s)):
            c1,c2=s[i],t[i]
            if c1 not in l:
                if c2 in l.values():
                    return False
                l[c1]=c2
            elif l[c1]!=c2:
                return False
        return True


# Running in terminal/console:
if __name__ == '__main__':
    Instant = Solution()
    Solve = Instant.isIsomorphic(s = "egg", t = "add")
    # s = "egg", t = "add" -> true
    # s = "foo", t = "bar" -> false
    print(Solve)

File: LeetCode/Java Solutions/Isomorphic Strings/isomorphic.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        int[] mapS = new int[256];
        int[] mapT = new int[256];
        for (int i = 0; i < s.length(); i++) {
            if (mapS[s.charAt(i)] != mapT[t.charAt(i)]) return false;
            mapS[s.charAt(i)] = i + 1;
            mapT[t.charAt(i)] = i + 1;
        }
        return true;
    }
}